pulsar-io/mongo/src/main/java/org/apache/pulsar/io/mongodb/MongoSourceConfig.java

```diff
@@ -0,0 +1,102 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.io.mongodb;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.experimental.Accessors;
+import org.apache.pulsar.io.core.annotations.FieldDoc;
+
+/**
+ * Configuration class for the MongoDB Source Connectors.
+ */
+@Data
+@EqualsAndHashCode(callSuper = false)
+@Accessors(chain = true)
+public class MongoSourceConfig extends MongoAbstractConfig {
+
+    private static final long serialVersionUID = 1152890092264945317L;
+
+    public static final SyncType DEFAULT_SYNC_TYPE = SyncType.INCR_SYNC;
+
+    public static final String DEFAULT_SYNC_TYPE_STR = "incr";
+
+    @FieldDoc(
+            defaultValue = DEFAULT_SYNC_TYPE_STR,
+            help = "The message synchronization type of the source connector. "
+                    + "The field values can be of two types: incr and full. "
+                    + "When it is set to incr, the source connector will only watch for changes made from now on. "
+                    + "When it is set to full, the source connector will synchronize currently existing messages "
+                    + "and watch for future changes."
+    )
+    private SyncType syncType = DEFAULT_SYNC_TYPE;
+
+    public static MongoSourceConfig load(String yamlFile) throws IOException {
+        final ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+        final MongoSourceConfig cfg = mapper.readValue(new File(yamlFile), MongoSourceConfig.class);
+
+        return cfg;
+    }
+
+    public static MongoSourceConfig load(Map<String, Object> map) throws IOException {
+        final ObjectMapper mapper = new ObjectMapper();
+        final MongoSourceConfig cfg =
+                mapper.readValue(new ObjectMapper().writeValueAsString(map), MongoSourceConfig.class);
+
+        return cfg;
+    }
+
+    /**
+     * Override syncType setter method.
+     *
+     * @param syncType Sync type string.
+     */
+    public void setSyncType(String syncType) {
```

Comment:
@315157973 
Functionally, I think that using a no-argument constructor and a `setter` method as same as using `@JsonCreater`. So I overrided the `setter` method of `syncType`. 
Using `@JsonCreator` can express more clearly. But it will cause `MongoAbstractConfig`, `MongoSourceConfig`, and `MongoSinkConfig` to add constructors with arguments (Because the `@Data` annotation only adds a no-argument constractor).

BTW, is this deserialization method OK?
```java
this.syncType = SyncType.valueOf(syncTypeStr.toUpperCase());
```